server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -161,6 +162,17 @@ public static String convertToJson(BytesReference bytes, boolean reformatJson, X
         return convertToJson(bytes, reformatJson, false, xContentType);
     }
 
+    /**
+     * Accepts a JSON string and reformats it. This is useful where a piece of JSON is formatted for legibility, but needs
+     * to be stripped of unnecessary whitespace e.g. for comparison in a test.
+     * @param json the JSON to format
+     * @return reformatted JSON
+     * @throws IOException if the reformatting fails, e.g. because the JSON is not well-formed
+     */
+    public static String reformatJson(String json) throws IOException {
```

Comment:
I think a name that better reflects what the `reformat` actually does would be better, something like `stripWhitespace` maybe ?

Reply:
Thanks @atorok, I renamed the method as you suggested.